### PR TITLE
SOS-130 Messages should clear

### PR DESCRIPTION
**Pull request by SeanTConrad**
Changes include:

- Unread messages are bold
- Clicking calls API to mark messages as read
- Pointer cursor over message to mark as clickable
- Messages sorted newest to oldest
- Badge now shows just number of unread

### CODE REVIEW

**Comment by SeanTConrad** [comment]
There's still some bugs with Messages. Some old messages don't have great data (null created at). 

The linked sharing page is a bit confusing. 

There's no place to see all historic messages. And messages are never deleted.

But it's good enough for demo purposes and better than it was before this ticket.

File: client/src/components/AlertMessage.tsx [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-
 export default function AlertMessage(props:Props){
     const { message } = props;
+
+    const {  markMessageRead } = useMessages()
+
+    const [ isRead, setRead ] = useState(message.is_read);
+
     const [ expanded, expand, collapse, flip ] = useBoolean(false);
     const to='You',
```
Comment (aylusltd) [inline client/src/components/AlertMessage.tsx:26]: This doesn't need to be changed, but there should be a message.recipient and message.recipient_org if you want to use them.

File: client/src/components/NetWorth.tsx [hunk 2]
```diff
@@ -127,18 +127,25 @@ export default function NetWorth(props: Props) {
...
+                            <Asset companyId={props.companyId} canEdit={props.canEdit}/>
                             </div>
+
                             <div className="data">
-                            <p className="title">Assets</p>
+                            {/* 3 columns */}
+                            <h4 className="holdingsHeading">Assets</h4>
+                            <p>{''}</p>
+                            <p>{''}</p>
+                            <p className="dataItem">Cash</p>{' '}
```
Comment (aylusltd) [inline client/src/components/NetWorth.tsx:143]: Should this be hardcoded? We can read it from the account types.

File: client/src/components/Alerts.tsx [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
+import { ExpandedMessageType } from './types';
+import { isBefore, parseISO } from 'date-fns';
 
 
 export default function Alerts() {
-    const theme=useTheme();
     const { userState, getCurrentUser } = useCurrentUser();
     const { messageState, getMessages } = useMessages()
 
+    const [ messages, setMessages ] = useState<ExpandedMessageType[]>([]);
```
Comment (aylusltd) [inline client/src/components/Alerts.tsx:15]: Why two message states?

**Review by aylusltd — CHANGES_REQUESTED** [review]

File: client/src/components/NetWorth.tsx (hunk 2 above)
Comment (SeanTConrad) [reply client/src/components/NetWorth.tsx:143]: For NetWorth, I only fixed a bug where they styling for the version without liabilities didn't match the version with. This is all the old, original code. I will add a ticket in the backlog to make this MUI, more typed, etc.

File: client/src/components/AlertMessage.tsx (hunk 1 above)
Comment (SeanTConrad) [reply client/src/components/AlertMessage.tsx:26]: Both "You" and "Me" were hardcoded when I started. I didn't know if this list ever showed my sent messages. If it always just show my inbox, "You" is very clean from a UI perspective.

File: client/src/components/Alerts.tsx (hunk 3 above)
Comment (aylusltd) [reply client/src/components/Alerts.tsx]: This is still in the PR, did you push that?

**Pull request merged**